python/sglang/srt/layers/quantization/gptq.py

```diff
@@ -0,0 +1,412 @@
+import logging
+from fractions import Fraction
+from typing import Any, Dict, List, Optional, Union
+
+import torch
+from vllm.model_executor.layers.quantization.gptq import GPTQLinearMethod
+from vllm.model_executor.layers.quantization.gptq_marlin import (
+    GPTQMarlinLinearMethod,
+    GPTQMarlinMoEMethod,
+)
+from vllm.model_executor.layers.quantization.marlin import MarlinLinearMethod
+from vllm.model_executor.layers.quantization.utils.marlin_utils import (
+    check_marlin_supported,
+)
+from vllm.platforms import current_platform
+from vllm.scalar_type import scalar_types
```

Comment:
@yinfan98  These are not `extra` vllm depends.  SGLang already imports them except indirectly via calls to vLLM GPTQ/Marlin code paths. Meaning, they are always imported when GPTQ are invocked. Here, I am migrating the GPTQ related Config classes here so there is no dependency on vllm 0.7.3. SGlang only synced to vllm 0.7.2.